File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    convert_cnt = zero_cnt = 0
    while 1:
        convert_cnt += 1
        origin = len(s)
        s = len(s.replace("0", ""))
        zero_cnt += origin - s
        
        
        s = format(int(s), "b")
        
        if s == "1":
            return [convert_cnt, zero_cnt]

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
const solution = s => {
    let procedureCnt = 0
    let zeroCnt = 0
        
    const makeOne = s => {
        if (s === "1") return
        else {
            let newS = ""
            for (const c of s){
                if (c === "1") newS += c
                else zeroCnt++
            }
            procedureCnt++
            makeOne(newS.length.toString(2))
        }
    }
    makeOne(s)
    return [procedureCnt, zeroCnt]
}